applications/ShallowWaterApplication/python_scripts/coupling/read_from_sw_interface_process.py

```diff
@@ -0,0 +1,344 @@
+
+        KM.Process.__init__(self)
+        self.settings = settings
+        self.settings.ValidateAndAssignDefaults(self.GetDefaultParameters())
+
+        self.interface_model_part = model[self.settings["interface_model_part_name"].GetString()]
+        self.input_model_part = model.CreateModelPart(self.settings["input_model_part_name"].GetString())
+
+        self.interpolate_sw_to_pfem_utility = SW.InterpolateSwToPfemUtility()
+        
+        self.interval = KM.IntervalUtility(self.settings)
+        self.variables = GenerateVariableListFromInput(self.settings["list_of_variables"])
+        self.variables_to_fix = GenerateVariableListFromInput(self.settings["list_of_variables_to_fix"])
+        
+        self.hdf5_import = import_model_part_from_hdf5_process.Factory(self._CreateHDF5Parameters(), model)
+        
+        self.hdf5_process = single_mesh_temporal_input_process.Factory(self._CreateHDF5Parameters(), model)
+        self._GetInputTimes(self.settings['file_settings'])
+
+
+    def Check(self):
+        '''Check the processes.'''
+        self.hdf5_import.Check()
+        self.hdf5_process.Check()
+        free_surface_is_present = False
+        height_is_present = False
+        for variable in self.variables:
+            if variable == SW.FREE_SURFACE_ELEVATION:
+                free_surface_is_present = True
+            if variable == SW.HEIGHT:
+                height_is_present = True
+        if free_surface_is_present and not height_is_present:
+            self.variables.append(SW.HEIGHT)
+
+
+    def ExecuteInitialize(self):
+        '''Read the input_model_part and set the variables.'''
+        self.hdf5_import.ExecuteInitialize()
+        self._CheckInputCoordinates()
+        self.FindPfemHeight() 
+        self._CheckInputVariables()
+        
+
+    def ExecuteInitializeSolutionStep(self):
+        '''Set the variables in the input_model_part at the current time.'''
+        current_time = self.interface_model_part.ProcessInfo.GetValue(KM.TIME)
+        if self.interval.IsInInterval(current_time):
+            self._SetCurrentTime()
+            self.hdf5_process.ExecuteInitializeSolutionStep()
+            self.ComputeAverageValues()
+            self._CheckInputVariables()
+            self.DistributeVelocityToPfem()
+        else:
+            if self.settings["default_time_after_interval"] is not None:
+                self._SetDefaultTime()
+                self.hdf5_process.ExecuteInitializeSolutionStep()
+                self.ComputeAverageValues()
+                self._CheckInputVariables()
+                self.DistributeVelocityToPfem()
+                self._SmoothDefaultValue()
+
+    def ComputeAverageValues(self):
+        if self.interface_model_part.ProcessInfo[KM.DOMAIN_SIZE] == 2:
+            avg_vel_x = 0
+            avg_h = 0
+            avg_vel_z = 0
+            n = 0
+            for node in self.input_model_part.Nodes:
+                n = n+1
```

Comment:
node loops in python are really slow... we should implement this in c++ ASAP

Reply:
Ok, I will do it ASAP, thanks for the suggestion